### DISCUSSION TITLE
beamdyn coordinate

### DISCUSSION THREAD

**Discussion by Unique-Wang** [OpenFAST/openfast#1018] [Q&A] [answered]
Dear @jjonkman ,

I've been reading this document http://forums.nrel.gov/t/simulations-using-beamdyn/2294/6 and I'm a bit confused by the **point O**,
![C5TQCR$Z_7{L40T_J%1%0`E](https://user-images.githubusercontent.com/87693817/155987316-7fc64ae3-1c85-4845-94cd-e53639865f89.png),I know its the origin of  the cross section (defined by the reference axis) but i don't think I fully understand this like ,if I already know the distance of **1/2*chord** to the **blade pitch axis** in **x** direction and **y** direction , then how do I get **kp_xr** and **kp_yr**? Whats the relationship between **1/2*chord** and **point O**? I'd appreciate it if you could help me with this!

Best regards,

**Comment by jjonkman** [comment, marked as answer]
Dear @Unique-Wang,

The reference axis specified in BeamDyn is user defined.  It should generally follow the geometry of the blade (with precurve, presweep), but it need not be located at half chord.  Some users prefer specifying the reference axis as the loci of shear center, tension center, or mass center, but the reference axis need not be any of these explicitly.

Best regards,

**Reply by Unique-Wang** [reply]
Thank you a lot!
And another thing I want to make sure is when I use BeamDyn for Blade in my model instead of ElastoDyn for blade, I found that  in **ed.sum**, **Rotor Mass** doesn't include  **3*Blade Mass**  anymore， is that because something wrong with the settings or it's just supposed to be like that?

Best regards,

**Reply by jjonkman** [reply]
This is expected behavior.  The ElastoDyn summary file only contains properties of the ElastoDyn module.  When the blades are modeled in BeamDyn, within the rotor, only the hub is accounted for in ElastoDyn, and so, the blade and rotor mass and inertia reported in the ElastoDyn summary file only accounts for the hub.

Best regards,

**Reply by Unique-Wang** [reply]
Thank you, that’s pretty clear to me now.
And every time I try  to add **nodal outputs section** to **BeamDyn primary input file**,  I always encounter the problem that happens right after  **TStart**  as is shown in the figure
![企业微信截图_16467083376036](https://user-images.githubusercontent.com/87693817/157157682-d681fd4c-22e4-4fff-9827-cdec5794bd07.png)

Best regards,

**Reply by jjonkman** [reply]
Perhaps this is the same issue that was reported on our forum: https://forums.nrel.gov/t/sending-rtaeropwr-via-servodyn-to-bladed-style-dll-controller/2043/15 and fixed via PR # https://github.com/OpenFAST/openfast/pull/996?

**Reply by Unique-Wang** [reply]
Thank you !  It seems that this bug is already fixed in the latest version of  OpenFAST, because the same error doesn't occur anymore when  I use  version3.1. 

Best regards,

**Reply by Unique-Wang** [reply]
And I encountered this error, 
![企业微信截图_16469028377579](https://user-images.githubusercontent.com/87693817/157632361-3c2458eb-ceb2-44c5-b627-5af15493b901.png), in #1007 you said this is because of the incompatibility in #1007, but I'm really confused  by  'Closest distance is **5.81670E-02** m'

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

These errors are notifying you of an incompatibility between the AeroDyn and BeamDyn meshes.  5.81670E-02 m is the size of the mismatch.  All AeroDyn nodes must be able to project orthogonally onto the Line2 elements of the BeamDyn mesh.  See how the Line2-to-Line2 mesh mapping search is described in our AIAA SciTech 2014 paper: https://www.nrel.gov/docs/fy14osti/60742.pdf.

Best regards,

**Reply by Unique-Wang** [reply]
Thanks a lot!
And one more thing that I've been confused so long is **solution doesn't converge**, I know it has something to do with **DT**, and I set **DT** ten(or even over ten) times smaller(0.0005s) than before(0.005s) to make the solution converge, but this slows down the calculation so much.I don't know if it's appropriate to ask but,   is 0.0005s normal?
![1647174618(1)](https://user-images.githubusercontent.com/87693817/158059391-e29136c8-40b4-41f1-89c0-d6d849bda0ff.png)

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

We are aware that BeamDyn requires smaller time steps than would generally be desired.  We have plans to address this issue by introducing a tight coupling algorithm between BeamDyn and ElastoDyn (and between ElastoDyn and SubDyn).  Until then, the time step is likely driven by high natural frequencies in your BeamDyn model that are being propagated into the ElastoDyn model through the structure-to-structure coupling.

Are the stiffness and mass properties you've specified in your BeamDyn model physically realistic?  If the mass is artificially low or the stiffness is artificially high, then unphysically high frequencies could be introduced, requiring small time steps, which could be increased by introducing physically accurate mass and stiffness values.  Reducing the order of the finite element in BeamDyn may also allow you to use larger time steps.

Best regards,

**Comment by Unique-Wang** [comment]
Dear @jjonkman ,

I recently found that when I use beamdyn for blades, the simulation results of my model basically match my Bladed model results at low wind speed(such as 6.5m/s), but there would always be severe fluctuations of rotor speed and generator power, and then the controller would shut the model down if I use a relatively higher wind speed (such as 12.5m/s and 15m/s) . But this would not happen if I use  elastodyn for blades.
![image](https://user-images.githubusercontent.com/87693817/160341626-a6a569f5-486e-4767-aa51-e669ac38d10f.png)
What could be the reason you think?

Best regards,

**Reply by jjonkman** [reply]
Dear @Unique-Wang,

I would guess that your BeamDyn model is capturing blade flexural modes that are not accounted for in your ElastoDyn model, such as torsion, or higher flapwise and bending modes, and that these modes are interacting with the controller behavior.  It could also mean that your BeamDyn model has unstable modes at the higher wind speeds.

Best regards,
